src/events.rs

```diff
@@ -522,4 +522,36 @@ mod tests {
         assert_eq!(events[0].1.variant_index, encoded_event[0]);
         assert_eq!(events[0].1.data.0, encoded_event[1..]);
     }
+
+    #[test]
+    fn event_containing_explicit_index() {
+        #[derive(Clone, Encode, TypeInfo)]
+        #[repr(u8)]
+        #[allow(trivial_numeric_casts)] // required because the Encode derive produces a warning otherwise
```

Comment:
```suggestion
        #[allow(trivial_numeric_casts, clippy::unnecessary_cast)] // required because the Encode derive produces a warning otherwise
```

Reply:
@sander2 just do this, looks like clippy is wrong it's just a test so should be ok :)